package cache;

message proWarrior{
    required uint32 sid = 1;             //单位编号
    required uint32 level = 2;           //单位等级
    required uint32 exp = 3;             //经验值
    required uint32 up = 4;              //突破等级
    required uint32 attack_add = 5;      //攻击培养
    required uint32 defense_add = 6;            //防御培养
    required uint32 hp_add = 7;                 //血量培养
    required uint32 potential = 8;              //潜力
    required uint32 skill_exp = 9;            //技能经验
    required uint32 skill_lv = 10;            //技能等级
    required uint32 grade = 11;             //弟子评价
    required uint32 develop_item_num = 12;  //培养单数量
    required uint32 dress_id = 13;      //时装编号
}

//获得单个单位 返回proWarriorRes
message proGetWarrior{
    required uint32 user_id = 1;                //玩家编号
    required uint32 sid = 2;             //单位编号
}

//获得全部单位 返回proWarriorsRes
message proGetWarriors{
    required uint32 user_id = 1;                //玩家编号
    repeated uint32 sids = 2;                   //弟子编号
}

//添加单位 返回proRes
message proAddWarrior{
    required uint32 user_id = 1;                //玩家编号
    required proWarrior warrior = 2;            //单位
}

//更新单位 返回proRes
message proUpdateWarrior{
    required uint32 user_id = 1;                //玩家编号
    required proWarrior warrior = 2;            //单位
}

//批量更新单位 返回proRes
message proUpdateWarriors{
    required uint32 user_id = 1;                //玩家编号
    repeated proWarrior warriors = 2;           //单位列表
}

//删除单位 返回proRes
message proRemoveWarrior{
    required uint32 user_id = 1;                //玩家编号
    required uint32 sid = 2;                    //单位编号
}

//删除多个单位 返回proRes
message proRemoveWarriors{
    required uint32 user_id = 1;                //玩家编号
    repeated uint32 sids = 2;                   //单位编号列表
}

//单位魂魄
message proWarriorSoul{
    required uint32 sid = 1;                    //单位编号
    required uint32 soul_num = 2;               //价值魂魄数量
}

//分解单位 返回proRes
message proDecomposeWarrior{
    required uint32 user_id = 1;                //玩家编号
    required proWarriorSoul warrior = 2;        //分解的单位
}

//分解单位 返回proRes
message proDecomposeWarriors{
    required uint32 user_id = 1;                //玩家编号
    repeated proWarriorSoul warriors = 3;       //分解的单位列表
}

//包含单位的结果
message proWarriorRes{
    required uint32 code = 1;                   //错误码
    optional proWarrior warrior = 2;            //单位信息
}

//包含多个单位的结果
message proWarriorsRes{
    required uint32 code = 1;                   //错误码
    repeated proWarrior warriors = 2;            //单位信息
}
